### DISCUSSION TITLE
Support Twitch's new tool for for saying hello

### DISCUSSION THREAD

**Discussion by killertofus** [Chatterino/chatterino2#3494] [Ideas]
https://twitter.com/TwitchSupport/status/1481008097749573641

**Comment by Mm2PL** [comment]
The one that just breaks webchat? If they can't support it properly or rather at all, why should we? Also sending these is probably going to be impossible because 1p APIs. Update: it no longer breaks webchat.

**Comment by zneix** [comment]
I recently used this feature myself in webchat and introduced myself to one streamer; couldn't find any relevant PubSub topics and the introduction looked like a normal chat message to me. But maybe it was the fact that feature is in some rollout, where I don't see it properly in chat(?). Not sure, but currently there's no way for us to even start working on adding this.
~~Edit: I lied - this is apparently sent to us as a regular chat message, but it has special msg-id: `msg-id=user-intro` ([raw irc message](https://kotmisia.pl/haste/raw/WmHxnMM1pnSNp82))~~ nevermind, while I was editing mm made his comment ;p

And... I don't know if we even want this, perhaps other people could give their input, I'm neutral myself.

**Comment by Mm2PL** [comment]
Those messages are received through IRC.
```
@badge-info=;badges=glhf-pledge/1;color=#DAA520;display-name=Mm2PL;emotes=301512758:46-53;first-msg=1;flags=;id=1c49cf10-7c26-4886-932d-cb966329ea0e;mod=0;msg-id=user-intro;room-id=[expunged];subscriber=0;tmi-sent-ts=1642018534517;turbo=0;user-id=117691339;user-type= :mm2pl!mm2pl@mm2pl.tmi.twitch.tv PRIVMSG #[expunged] :Hi I'm here to see how this new feature works pajaDent
```

The crucial part of this is `@msg-id=user-intro`. Also it's worth noting that `@first-msg=1` and this new introduction aren't equivalent because you can chat before introducing yourself.
